pkg/build/webhook/generic/generic.go

```diff
@@ -65,8 +66,13 @@ func (p *WebHookPlugin) Extract(buildCfg *api.BuildConfig, secret, path string,
 		}
 
 		var data api.GenericWebHookEvent
+		body, err = yaml.ToJSON(body)
```

Comment:
it does, but the error is really confusing ("yaml: unmarshal errors...")... we should only do it when the content-type indicates it is yaml